buffer.go

```diff
@@ -33,6 +33,37 @@ func SingleTagBuffer(key string, value interface{}) *Buffer {
 	return b
 }
 
+// BuildBuffer is used to build a *Buffer that contains an arbitrary number
+// of tags. Sample usage:
+//
+//	bld := BuildBuffer()
+//	bld.Add("a", 1)
+//	bld.Add("b", 2)
+//	buf := bld.Finish()
```

Comment:
Could you add a comment somewhere explaining why this exists and why it's more efficient to use than the alternative (repeat calls to `Buffer.Add`)?

Reply:
Done.